0023-backend-v3.md

```diff
@@ -0,0 +1,239 @@
+# BackendV3
+
+| **Status**        | **Proposed/Accepted/Deprecated** |
+|:------------------|:---------------------------------------------|
+| **RFC #**         | ####                                         |
+| **Authors**       | Elena Peña Tapia (ept@zurich.ibm.com), Matthew Treinish (matthew.treinish@ibm.com)   |
+| **Deprecates**    | RFC that this RFC deprecates                 |
+| **Submitted**     | YYYY-MM-DD                                   |
+| **Updated**       | YYYY-MM-DD                                   |
+
+## Summary
+Backends are the "de facto" abstractions for quantum devices in the Qiskit ecosystem, extending beyond a simple combination of their components. The current Qiskit backend model, however, clashes with the execution model of primitives. This is a proposal to add a new `BackendV3` class that fills the gap in the current Qiskit backend + execution model.
+
+> Instead of a required `backend.run()`, the proposal for `BackendV3` implementations is to offer **primitive factory methods**: `backend.sampler()` and `backend.estimator()` that return their respective instantiated primitive classes (details TBD), as well as an optional `backend.run()` for 3rd party and local simulation backends (for backwards compatibility and testing, up to discussion).
+
+For example, for `qiskit-ibm-runtime`, we could have:
+
+```python
+from qiskit_ibm_runtime import QiskitRuntimeService
+
+service = QiskitRuntimeService()
+backend = service.backend("ibm_fez")
+estimator = backend.estimator(mode="batch", options={"shots":1024})
+```
+
+instead of:
+```python
+from qiskit_ibm_runtime import QiskitRuntimeService, Estimator
+
+service = QiskitRuntimeService()
+backend = service.backend("ibm_fez")
+estimator = Estimator(backend=backend, mode="batch", options={"shots":1024})
+```
+
+## Motivation
+
+### Why are we doing this? (With a bit of History)
+
+The Qiskit `Backend` object was originally introduced to provide an abstract model of a **physical quantum computer**, enabling software to interact with it. There are two key types of interaction with a `Backend`:
+
+1. **Transpilation** -> "we need to access hardware constraints to optimize circuits (to run on a QPU)"
+2. **Execution** -> "we need an interface for sending circuits to the QPU and receiving an output"
+
+The original `Backend` interface (`BackendV1`) was designed primarily for circuit execution, the most fundamental ("basic") interaction with quantum computers. However, the need for improved transpilation support led to the introduction of `BackendV2`, and incorporating an abstraction for the "QPU", the `Target`. 
```

Comment:
The original backend interface was actually `BaseBackend` (i.e. `BackendV0`), which was basically `BackendV1` but with 2 key differences, `BaseBackend` took a `qobj` as an input instead of a `QuantumCircuit` on `.run()` and it was also explicitly tied to the IQX API formats. For `BackendV1` I made it explicitly decoupled from IBM's REST API payload format but still using the same format at the time, this was to decouple the Python code from the API format to allow changes on either side, and removed the use of qobj from the `run()`.

Reply:
Added in [0427d3d](https://github.com/Qiskit/RFCs/pull/77/commits/0427d3da30df0bf05fcf405b92120e4cbfa4edc5)